Split databases from Context

tavplubix opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Non-significant (changelog entry is not required)

Required for #7512

tavplubix committed:
SHOW CREATE for DatabaseMemory

tavplubix committed:
store temporary tables in DatabaseMemory

tavplubix committed:
add DatabaseCatalog

tavplubix committed:
move databases from Context to DatabaseCatalog

tavplubix committed:
fix build

tavplubix committed:
fixes

tavplubix committed:
move DDLGuard to DatabaseCatalog

tavplubix committed:
fix

tavplubix committed:
move ViewDependencies to DatabaseCatalog

tavplubix committed:
remove tryGetExternalTable

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
make Context::getLock() private

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
better get... methods

tavplubix committed:
less Context::getTable() usages

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
remove another method from Context

amosbird commented:
https://github.com/ClickHouse/ClickHouse/issues/9136 seems to be doable inside this PR

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
do not allow custom database for system tables

tavplubix committed:
add StorageID to ASTInsertQuery

tavplubix committed:
fixes

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
remove tryGetTable

tavplubix committed:
fix test_part_log_table

part_log must be created in system database,
because otherwise it is not guaranteed that part_log loaded
before it can be used by other table

tavplubix committed:
remove Context::getTable()

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
improve temporary tables

tavplubix committed:
fixes

tavplubix committed:
Merge branch 'master' into split_databases_from_context

tavplubix committed:
remove isExternalTableExist

tavplubix committed:
enable UUIDs for temporary tables

tavplubix committed:
fix build

tavplubix committed:
minor improvements

tavplubix committed:
add test for tmp tables

tavplubix committed:
fix

tavplubix committed:
fix
